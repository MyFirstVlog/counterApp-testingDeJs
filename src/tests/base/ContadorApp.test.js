import React from 'react';
import { shallow } from 'enzyme';
import ContadorApp from '../../CounterApp';


describe('Pruebas tecnicas de la app del super contador web', () => {

    let wrapper = shallow(<ContadorApp />)

    //Ciclo de vida, se inicializa a esta condicion en cada uno de los tests

    beforeEach(()=> {
        wrapper = shallow(<ContadorApp />)
    })
    
    test('verificacion del snapshot renderizado ', () => {


        expect(wrapper).toMatchSnapshot()
        
    })

    test('verificacion del valr esperado  ', () => {
        
        const wrapper = shallow(<ContadorApp value={7}/>)
        const defaultNumber = wrapper.find('h2').text()

        expect(defaultNumber).toBe('7')


    })

    test('incrementar con el boton de +1', () => {
        
        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text()

        expect(counterText).toBe('8')

        

    })

    test('decrementar con el boton de -1', () => {
       
        
        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text()

        expect(counterText).toBe('6')

    })
    
    test('debe colocar el valor por defecto', () => {
       
        
        const wrapper = shallow(<ContadorApp value={105}/>)

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text()

        expect(counterText).toBe('105')

        console.log(counterText)

    
    })

    
    
})

