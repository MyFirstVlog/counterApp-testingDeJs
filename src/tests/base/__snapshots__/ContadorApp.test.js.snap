// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas tecnicas de la app del super contador web verificacion del snapshot renderizado  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    id="h2Text"
  >
    7
  </h2>
  <button
    id="btn"
    onClick={[Function]}
  >
    + 1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
